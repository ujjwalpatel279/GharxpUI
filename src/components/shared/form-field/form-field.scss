// ========================================================================
// COMPONENTS/SHARED/FORM-FIELD
// ========================================================================

@import '~sassHelpers';

.c-form-field {
  margin-bottom: spacer(3);

  &__label {
    margin-bottom: spacer(1);
  }

  &__error {
    font-size: font-size(2);
    color: color(errorDark);
  }

  &__hidden {
    display: none;
  }
}
