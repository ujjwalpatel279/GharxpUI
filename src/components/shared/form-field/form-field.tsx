import React, { FunctionComponent } from 'react';

interface FormFieldProps {
  id?: string;
  label: string;
  error?: string;
  display?: boolean;
}

import './form-field.scss';

export const FormField: FunctionComponent<FormFieldProps> = ({ id, label, error, children, display }) => {
  const labelAttributes = {
    className: 'c-form-field__label',
    htmlFor: id || null,
  };

  return (
    <div className={display ?? true ? 'c-form-field' : 'c-form-field__hidden'}>
      <label {...labelAttributes}>{label}</label>
      {children}
      {error && (
        <p key={error} role="alert" className="c-form-field__error">
          {error}
        </p>
      )}
    </div>
  );
};
