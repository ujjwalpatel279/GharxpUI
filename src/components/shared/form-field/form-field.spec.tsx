import React from 'react';
import { render, cleanup } from '@testing-library/react';

import { FormField } from './form-field';

describe('FormField Component', () => {
  afterEach(cleanup);

  it('renders the form field component', () => {
    const { container } = render(<FormField label="testLabel">Form Field Component</FormField>);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('error message is shown', () => {
    const { queryByText } = render(
      <FormField label="testLabel" error="Test error message">
        Form Field Component
      </FormField>,
    );
    expect(queryByText('Test error message')).toBeInTheDocument();
  });
});
