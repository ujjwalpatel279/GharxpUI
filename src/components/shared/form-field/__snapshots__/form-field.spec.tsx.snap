// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormField Component renders the form field component 1`] = `
<div
  class="c-form-field"
>
  <label
    class="c-form-field__label"
  >
    testLabel
  </label>
  Form Field Component
</div>
`;
